File: Tipos de Dados/Cadeia de Caracteres/cadeiacaracteres.cpp
//C++
#include <iostream>
#include <string>

using namespace std;

int main() {
    string str1 = "Olá";
    string str2 = "Mundo";

    // Concatenar
    string concatenado = str1 + " " + str2;
    cout << "Concatenado: " << concatenado << endl;

    // Substring
    string sub = concatenado.substr(0, 3);
    cout << "Substring: " << sub << endl;

    // Comparar strings
    bool iguais = (str1 == str2);
    cout << "As strings são iguais? " << iguais << endl;

    // Tamanho da string
    cout << "Tamanho de str1: " << str1.length() << endl;
    
    return 0;
}

File: Tipos de Dados/Cadeia de Caracteres/cadeiacaracteres.rs
//Rust
fn main() {
    let str1 = "Olá";
    let str2 = "Mundo";

    // Concatenar
    let concatenado = format!("{} {}", str1, str2);
    println!("Concatenado: {}", concatenado);

    // Substring
    let sub = &concatenado[0..3];
    println!("Substring: {}", sub);

    // Comparar strings
    let iguais = str1 == str2;
    println!("As strings são iguais? {}", iguais);

    // Tamanho da string
    println!("Tamanho de str1: {}", str1.len());
}
